### DISCUSSION TITLE
How do I contribute to bevy as a novice?

### DISCUSSION THREAD

**Discussion by TirushOne** [bevyengine/bevy#11225] [Q&A]
So I really want to comtribute to bevy, but it's all a bit overwheling even trying to comprehend what the architecture of the engine is and how it all works. My usual learning style involve asking an ungodly number of clarifiing questions until I have a wholistic understand of what seperates thing A from thing B. Why are thing A and B seperated? Why is X, Y and Z necessary? And so on. This isn't feaible with bevy on any large scale because of the decentralised nature of the contributions/volanteer force. So is my only option to just burry my head in the code until I have enough of an understanding of what everything is to start working on it? And even then without feedback from someone more senior it's diffucult to verify my understanding of anything. If I could I would like to work on some of the rendering features (although this is not my area) but I am happy to work on most anything that is needed  on the main bevy engine if I can have someone or a group of people with more expierence then me to answer my questions because I don't want to just flood the discssions section.

I understand there is a tag for issues for new comers to takle but I am asking because I want to know if there is a better path way to working of more significant engine features then just figuring it all out on my own.

(there will be spelling mistakes I apologise)

**Comment by TimJentzsch** [comment]
I think one great way to start is by reviewing open pull requests.
This allows you to gradually become more familiar with the code base while making important contributions to the engine.

In Bevy, anyone can review pull requests and community reviews count towards merging the feature.

You can filter the PRs by the Rendering label to find things you are interested in.

If you want to start making contributions yourself, a good way to start is by improving the docs. This can also have a great impact on other contributors and users while being easier to get into.

Hope this helps!

**Comment by JMS55** [comment]
Hey, come join #rendering-dev on the bevy discord! If you open a thread, I, and other contributors, would be happy to answer whatever questions you have. That's how I got started contributing to bevy :). Don't feel afraid to ask lots of questions!

**Reply by TirushOne** [reply]
ok that's acctaully quite helpful. I'll give that a go :)
I just made my first pr an hour or so ago so I'm already warming up better then I expected tbh
